rpm/src/test/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessorTest.java

```diff
@@ -0,0 +1,134 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.bouncycastle.bcpg.ArmoredInputStream;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.bc.BcPGPPublicKeyRing;
+import org.eclipse.packager.rpm.RpmSignatureTag;
+import org.eclipse.packager.rpm.Rpms;
+import org.eclipse.packager.rpm.parse.InputHeader;
+import org.eclipse.packager.rpm.parse.RpmInputStream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class RpmFileSignatureProcessorTest {
+
+    private static final String SOURCE_FILE_PATH = "src/test/resources/data/org.eclipse.scada-0.2.1-1.noarch.rpm";
+    private static final String PRIVATE_KEY_PATH = "src/test/resources/key/private_key.txt";
+    private static final String PUBLIC_KEY_PATH = "src/test/resources/key/public_key.txt";
+    private static final String RESULT_FILE_PATH = "src/test/resources/result/org.eclipse.scada-0.2.1-1.noarch.rpm";
+    private static final String RESULT_DIR = "src/test/resources/result";
+
+    @Test
+    @Order(1)
+    public void testSigningExistingRpm() throws IOException, PGPException {
+        // Read files
+        final String passPhrase = "testkey"; // Do not change
+        File rpm = new File(SOURCE_FILE_PATH);
+        File private_key = new File(PRIVATE_KEY_PATH);
+        if (!rpm.exists() || !private_key.exists()) {
+            fail("Input files rpm or private_key does not exist");
+        }
+        // Init the signed RPM
+        File resultDirectory = new File(RESULT_DIR);
+        resultDirectory.mkdir();
+        File signedRpm = new File(RESULT_FILE_PATH);
+        signedRpm.createNewFile();
+
+        try (FileOutputStream resultOut = new FileOutputStream(signedRpm);
+                InputStream privateKeyStream = new FileInputStream(private_key)) {
+            // Sign the RPM
+            RpmFileSignatureProcessor.perform(rpm, privateKeyStream, passPhrase, resultOut);
+
+            // Read the initial (non signed) rpm file
+            RpmInputStream initialRpm = new RpmInputStream(new FileInputStream(rpm));
+            initialRpm.available();
+            initialRpm.close();
+            InputHeader<RpmSignatureTag> initialHeader = initialRpm.getSignatureHeader();
+
+            // Read the signed rpm file
+            RpmInputStream rpmSigned = new RpmInputStream(new FileInputStream(signedRpm));
+            rpmSigned.available();
+            rpmSigned.close();
```

Comment:
@ctron I know this is an old PR, but I see this isn't really handled properly. I will make a PR.